ReactApp/src/components/EditorToolbar.jsx

```diff
@@ -6,69 +6,69 @@ import {
 	BlockInspector,
 	BlockToolbar,
 	Inserter,
+	store as blockEditorStore,
 } from '@wordpress/block-editor';
-import { Popover } from '@wordpress/components';
-// import { Sheet } from 'react-modal-sheet';
-import { showBlockPicker } from '../misc/Helpers';
+import { useSelect } from '@wordpress/data';
+import { Button, Popover, ToolbarButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { close, cog } from '@wordpress/icons';
 
-const EditorToolbar = (props) => {
+const EditorToolbar = () => {
 	const [isBlockInspectorShown, setBlockInspectorShown] = useState(false);
+	const { isSelected } = useSelect((select) => {
+		const { getSelectedBlockClientId } = select(blockEditorStore);
+		const selectedBlockClientId = getSelectedBlockClientId();
+		return {
+			isSelected: selectedBlockClientId !== null,
+		};
+	});
 
-	let addBlockButton;
-	if (props.registeredBlocks.length === 0) {
-		// TODO: use the native inserter
-		addBlockButton = <Inserter />;
-	} else {
-		addBlockButton = <button onClick={showBlockPicker}>+</button>;
+	function openSettings() {
+		setBlockInspectorShown(true);
+	}
+
+	function onCloseSettings() {
+		setBlockInspectorShown(false);
 	}
 
 	return (
-		<div className="gbkit gbkit-editor-toolbar">
-			<div className="gbkit-editor-toolbar_toolbar-group">
-				{addBlockButton}
+		<>
+			<div className="gbkit gbkit-editor-toolbar">
+				<Inserter />
+
+				{isSelected && (
+					<div className="gbkit-editor-toolbar_toolbar-group">
+						<ToolbarButton
+							title={__('Open Settings')}
+							icon={cog}
+							onClick={openSettings}
+							className="gbkit-editor-toolbar_settings_icon"
+						></ToolbarButton>
```

Comment:
For the pursuit of brevity, we might consider self closing this element, given it has no children. 
```suggestion
						/>
```